synapse: update to 1.21.2.

freshprince opened the pull request.

ahesford reviewed (changes requested):
We tend to ignore version restrictions on python packages unless there is a tight coupling between the packages. Also, restrictions in these templates are not consistent. Upstream `synapse` restricts versions on every package in `install_requires`, but only `python3-canonicaljson` is restricted here.

Please drop version restrictions in both templates. Also, please confirm that `python3-six` is necessary for `python3-canonicaljson`. I see no mention of it in the upstream `install_requires`.

freshprince commented:
Thanks for taking a look.

I removed the version restrictions and tested whether `canonicaljson` loads without `six` installed and it seemed to work. So I removed that dependency too.
